.rhdh/scripts/install-registry-proxy-images.sh

```diff
@@ -16,164 +16,185 @@
+    exit 1
 fi
 
-# Create the working directory
-mkdir -p "$WORKDIR"
-
-# Step 1: Pull the IIB image and extract catalog information
-echo "Pulling IIB image and extracting catalog information..."
-skopeo copy --all "docker://$IIB_IMAGE" "oci:$WORKDIR/iib"
-
-# Unpack the OCI image
-cd "$WORKDIR"
-echo "Listing contents of blobs/sha256/ for debugging..."
-ls -R "$WORKDIR/iib/blobs/sha256/"
-
-# Inspect index.json to locate manifests
-indexJson="$WORKDIR/iib/index.json"
-if [[ -f $indexJson ]]; then
-    manifests=$(jq -r '.manifests[].digest' "$indexJson")
-    echo "Found the following manifests: $manifests"
-    
-    images=()
-    
-    for manifest in $manifests; do
-        echo "Checking manifest: $manifest"
-        manifestDir="$WORKDIR/iib/blobs/sha256/${manifest#*:}"
-        
-        manifestJson="$manifestDir/manifest.json"
-        if [[ -f $manifestJson ]]; then
-            layers=$(jq -r '.[].layers[].digest' "$manifestJson")
-            config=$(jq -r '.[].config.digest' "$manifestJson")
-            images+=($layers $config)
+infof "[INFO] Using internal registry: ${INTERNAL_REGISTRY}"
+
+# Authenticate using the external route for image registry
+infof "[INFO] Authenticating to the external registry route..."
+REGISTRY_URL="${INTERNAL_REGISTRY}"  # Remove the https:// part
+oc registry login --registry=${REGISTRY_URL} --to=/tmp/pull-secret.json
+
+# Pull and push images to the internal registry
+for IMAGE in "${IMAGES[@]}"; do
+    # Convert to quay.io equivalent if needed
+    QUAY_IMAGE="${IMAGE/registry.redhat.io/quay.io}"
+
+    infof "[INFO] Checking if image ${IMAGE} exists..."
+    if ! skopeo inspect docker://"${IMAGE}" &>/dev/null; then
+        infof "[INFO] Image ${IMAGE} not found in registry.redhat.io. Checking quay.io..."
+
+        # Try quay.io if not found in registry.redhat.io
+        if ! skopeo inspect docker://"${QUAY_IMAGE}" &>/dev/null; then
+            errorf "[ERROR] Image not found in quay.io or registry.redhat.io: ${QUAY_IMAGE}"
+            exit 2
         else
-            echo "[WARNING] No manifest.json found for $manifest"
+            IMAGE="${QUAY_IMAGE}"
         fi
-    done
-else
-    echo "[ERROR] Could not find index.json. Exiting."
-    exit 1
-fi
+    fi
 
-# Step 2: Convert image references to Quay.io equivalents
-echo "Converting image references to Quay.io equivalents..."
-quay_images=()
-for image in "${images[@]}"; do
-    quay_image="${REMOTE_REGISTRY}/$(basename ${image#sha256:})"
-    quay_images+=("$quay_image")
+    # Pull the image locally
+    infof "[INFO] Pulling image: ${IMAGE}"
+    skopeo copy docker://"${IMAGE}" dir:/tmp/"${IMAGE##*/}"
+
+    # Push to the airgapped cluster using registry-proxy
+    LOCAL_IMAGE="${INTERNAL_REGISTRY}/rhdh-operator/$(basename ${IMAGE})"  # Add your namespace here
+    infof "[INFO] Pushing image to cluster: ${LOCAL_IMAGE}"
+    skopeo copy --authfile /tmp/pull-secret.json dir:/tmp/"${IMAGE##*/}" docker://${LOCAL_IMAGE}
```

Comment:
probably want to make sure you're using --all when doing a skopeo copy to ensure all arches and layers are included.

Reply:
also could you skip a step and just

```
    skopeo copy --authfile /tmp/pull-secret.json docker://"${IMAGE}"  docker://${LOCAL_IMAGE}
```

Or is there a reason to pull the image locally to disk to then upload it?